master/buildbot/configurators/janitor.py

```diff
@@ -39,43 +40,67 @@ class LogChunksJanitor(BuildStep):
     name = 'LogChunksJanitor'
     renderables = ["logHorizon"]
 
-    def __init__(self, logHorizon):
+    def __init__(self, logHorizon=None, horizonPerBuilder=None):
         super().__init__()
         self.logHorizon = logHorizon
+        self.horizonPerBuilder = horizonPerBuilder
 
     @defer.inlineCallbacks
     def run(self):
-        older_than_timestamp = datetime2epoch(now() - self.logHorizon)
-        deleted = yield self.master.db.logs.deleteOldLogChunks(older_than_timestamp)
-        self.descriptionDone = ["deleted", str(deleted), "logchunks"]
+        if self.logHorizon is not None:
+            older_than_timestamp = datetime2epoch(
+                now() - self.logHorizon)
+            deleted = yield self.master.db.logs.deleteOldLogChunks(
+                older_than_timestamp=older_than_timestamp)
+            self.descriptionDone = ["deleted", str(deleted), "logchunks"]
```

Comment:
This should be just before return I think, because if both if conditions are true then we would set descriptionDone before step is actually finished.

Reply:
OK, now I see that only either logHorizon or horizonPerBuilder is not None, not both.